src/elona/draw.cpp

```diff
@@ -706,170 +716,168 @@ void create_pcpic(int cc, bool with_equipments)
 
 
 
-void initialize_map_chip()
+void initialize_map_chips(const MapChipDB& db)
 {
-    // TODO: this could be called multiple times outside of
-    // initialize_all_chips. Add a method to clear only map chip
-    // buffers from PicLoader so they can be loaded again, or keep
-    // all chips loaded at once.
-    DIM3(chipm, 8, 825);
-    if (map_data.atlas_number == 0)
+    std::vector<PicLoader::MapType> predefined_extents;
+    predefined_extents.resize(ChipData::atlas_count);
+    tinted_buffers.clear();
+
+    for (const auto& data : db.values())
     {
-        chipm(5, 233) = 0;
-        chipm(6, 233) = 0;
-        for (int cnt = 26; cnt < 33; ++cnt)
-        {
-            chipm(0, cnt) = 4;
-        }
-        chipm(0, 568) = 4;
-        chipm(0, 569) = 4;
-        chipm(0, 570) = 4;
-        for (int cnt = 99; cnt < 132; ++cnt)
-        {
-            chipm(0, cnt) = 7;
-        }
-        for (int cnt = 165; cnt < 198; ++cnt)
+        auto& atlas = chip_data.get_map(data.atlas);
+
+        atlas[data.id] = std::move(data);
```

Comment:
Judging from using `std::move` here, the passed `MapChipDB` becomes empty state after the for loop ends, is it intended behavior?  If intended, please add comment about that in either this function or declaration of `the_map_chip_db`.

Reply:
I removed it to be consistent with the other `init` functions.